remotesync.py

```diff
@@ -114,34 +112,32 @@ def sync_path(self, base_path, base_filename):
                         "Somehow the source path no longer existed.  This should never happen, bailing out of this "
                         "transfer.")
                     break
-
-                error("failed to sync files to remote, retrying.  exception was '%s'" % e)
+                debug(f"{self.__class__.__name__}: {e.stderr.strip()}")  # Debug level prints stderr
+                error("Failed to sync files to remote, retrying.  exception was '%s'" % e)
                 self.pessimistic_wait()
 
     def list_files(self, realpath):
         remote_path = self.get_remote_path(realpath)
-
+        cmd = ["ssh", self.RSYNC_HOST, "find", pipes.quote(remote_path), "-type", "f", "-print"]
+        debug(f"Running command: {' '.join(cmd)}")
         while True:
             try:
                 # remote path must be quoted, lest it be interpreted wrongly
                 # by the shell on the server side.
-                output = subprocess.check_output([
-                    "ssh", self.RSYNC_HOST, "find", pipes.quote(remote_path),
-                    "-type", "f", "-print"
-                ])
+                output = subprocess.run(cmd, check=True, capture_output=True, encoding='utf8').stdout
 
-                # Decode all output immediately and split it, remember that
-                # rtorrent is returning unicode strings to us so we need
-                # to create unicode strings so that they can be compared
-                # to the list of locally completed files.
-                remote_files = output.decode('UTF-8').rstrip().split("\n")
+                # Split all output, remember that rtorrent is returning unicode
+                # strings to us so we need to create unicode strings so that
+                # they can be compared to the list of locally completed files.
```

Comment:
Nice approach using `encoding` keyword argument.

Arguably this comment is now redundant.  I'd either remove it or just write something like `# The use of 'encoding' argument to subprocess.run ensures that output is a unicode string`.

Reply:
Removed comment.